composer/datasets/streaming/writer.py

```diff
@@ -90,7 +92,9 @@ def __init__(self,
                  dirname: str,
                  fields: List[str],
                  shard_size_limit: int = 1 << 24,
-                 compression: Optional[str] = default_compression) -> None:
+                 compression: Optional[str] = default_compression,
+                 remote: Union[ObjectStore, str, None] = None,
```

Comment:
nit: prefer `Optional[...]` over `Union[None, ...]`

```suggestion
                 remote: Optional[Union[ObjectStore, str]] = None,
```